src/main/java/com/aws/greengrass/componentmanager/ComponentManager.java

```diff
@@ -366,12 +366,16 @@ public void checkPreparePackagesPrerequisites(List<ComponentIdentifier> componen
                                 "Special components with docker artifacts must include the "
                                         + "%s download plugin in the deployment", DOCKER_MANAGER_PLUGIN_SERVICE_NAME));
                     }
-                    Image image = Image.fromArtifactUri(artifact.getArtifactUri());
-                    if (image.getRegistry().isEcrRegistry() && image.getRegistry().isPrivateRegistry()
-                            && !componentNames.contains(TOKEN_EXCHANGE_SERVICE_TOPICS)) {
-                        throw new MissingRequiredComponentsException(String.format(
-                                "Components with private ECR docker artifacts must include "
-                                        + "the %s plugin in the deployment", TOKEN_EXCHANGE_SERVICE_TOPICS));
+                    try {
+                        Image image = Image.fromArtifactUri(artifact.getArtifactUri());
+                        if (image.getRegistry().isEcrRegistry() && image.getRegistry().isPrivateRegistry()
+                                && !componentNames.contains(TOKEN_EXCHANGE_SERVICE_TOPICS)) {
+                            throw new MissingRequiredComponentsException(String.format(
+                                    "Components with private ECR docker artifacts must include the %s plugin "
+                                            + "in the deployment", TOKEN_EXCHANGE_SERVICE_TOPICS));
```

Comment:
Changed the message. For the logic needing to be in docker specific component though, there's an issue. The component is supposed to be moved out to a plugin and it won't be loaded at the time of performing this check, we will need some sort of mapping within nucleus for which component can fulfill the download requirement for pluggable schemes like docker so that we can do this validation even when the plugin isn't yet loaded. So I moved this to the ArtifactDownloaderFactory now, which is not the best still, but hides it from the component manager at least